src/app-layer-modbus.c

```diff
@@ -3107,6 +3127,154 @@ static int ModbusParserTest16(void) {
     UTHFreePackets(&p, 1);
     return result;
 }
+
+/** \test Checks if stream_depth is correct */
+static int ModbusParserTest17(void) {
+    AppLayerParserThreadCtx *alp_tctx = AppLayerParserThreadCtxAlloc();
+    Flow f;
+    TcpSession ssn;
+
+    int result = 0;
+
+    memset(&f, 0, sizeof(f));
+    memset(&ssn, 0, sizeof(ssn));
+
+    f.protoctx  = (void *)&ssn;
+    f.proto     = IPPROTO_TCP;
+
+    StreamTcpInitConfig(TRUE);
+
+    SCMutexLock(&f.m);
```

Comment:
Please fix these tests to be:
1. compliant with FAIL_IF* logic (+ no conditions)
2. use FLOWLOCK_* instead of SCMutexLock for flow lock